### DISCUSSION TITLE
Importing icons based on type declarations

### DISCUSSION THREAD

**Discussion by david8z** [unocss/unocss#1480] [Q&A] [answered]
Was wondering if there was any possibility of importing icons based on type declarations.

```javascript
// types.ts
export type IconTypes = 'i-gg-profile' | 'i-gg-shopping-bag' | 'i-ph-star-bold'

// Component.vue
<script setup lang="ts">
import { IconTypes } from './types'  
const { icon } = defineProps<IconTypes>()
</script>

<template>
    <div :class="[icon]" />
</template>

```

Based on the example ☝️ it would be nice if the icons where dynamically imported.

**Comment by userquin** [comment, marked as answer]
@david8z add `// @unocss-include` to the types.ts module: https://github.com/unocss/unocss#scanning

**Reply by david8z** [reply]
Didn't knew about this ❤️  thanks a lot

**Reply by userquin** [reply]
@david8z Mark the discussion as answered

**Reply by david8z** [reply]
How could I do that?
